src/store/msgsSlice.ts

```diff
@@ -0,0 +1,80 @@
+import type { Page } from '@gnosis.pm/safe-react-gateway-sdk'
+import type { AddressEx } from '@gnosis.pm/safe-react-gateway-sdk'
+import type { Middleware } from '@reduxjs/toolkit'
+
+import { msgDispatch, MsgEvent } from '@/services/msg/msgEvents'
+import { isMessageListItem } from '@/utils/message-guards'
+import { makeLoadableSlice } from '@/store/common'
+import { selectPendingMsgs } from '@/store/pendingMsgsSlice'
+import type { RootState } from '@/store'
+
+// TODO: Export to gateway SDK
+export enum MessageListItemType {
+  DATE_LABEL = 'DATE_LABEL',
+  MESSAGE = 'MESSAGE',
+}
+
+export enum MessageStatus {
+  NEEDS_CONFIRMATION = 'NEEDS_CONFIRMATION',
+  CONFIRMED = 'CONFIRMED',
+}
+
+export type MessageDateLabel = {
+  type: MessageListItemType.DATE_LABEL
+  timestamp: number
+}
+
+export type Message = {
+  type: MessageListItemType.MESSAGE
+  messageHash: string
+  status: MessageStatus
+  logoUri: string // Implementation may change
+  name: string // Implementation may change
+  message: string | Record<string, unknown>
+  creationTimestamp: number
+  modifiedTimestamp: number
+  confirmationsSubmitted: number
+  confirmationsRequired: number
+  confirmations: {
+    owner: AddressEx
+    signature: string
+  }[]
+  proposedBy: AddressEx
+  preparedSignature: string | null // Name subject to change
+}
+
+export type MessageListPage = Page<MessageDateLabel | Message>
+
+const { slice, selector } = makeLoadableSlice('msgs', undefined as MessageListPage | undefined)
```

Comment:
> msgs

Please rename it to something more readable.

Reply:
Every instance of `msg`/`msgs` variable(s) or just the slice? We shorten transactions to `tx`, hence my reasoning.

Reply:
Just the slice name, because that's what you see in the Redux debugger.

Reply:
The word messages in general is too broad for this feature, in my opinion. It's unfortunate that it's now spread across the code.

Reply:
I've adjusted it in the open EIP-1271-related PR [here](https://github.com/safe-global/web-core/pull/1224/commits/09db7c73e7ccff9d526723d37ce7b44e7be9858a).

Reply:
> The word messages in general is too broad for this feature, in my opinion. It's unfortunate that it's now spread across the code.

It'ss renamable but it is/was the agreed upon term. Do you have other suggestions?